<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <param name="robot_description"
    textfile="$(find sensor_driver)/urdf/multivelodyne_3d.urdf" />
  <param name="publish_frequency" value="50." />
    
  <node name="robot_state_publisher" pkg="robot_state_publisher"
    type="robot_state_publisher" />
    
    
  <arg name="rviz" default="true" />
  <arg name = "online_mode" default = "false"/>

  
  <arg name="configfile" default="$(find sensor_driver)/config/config-RAY.xml" />
  <arg name="logdir" default="--log_dir=/home/zzh/catkin_ws/data/log" />
  <arg name="glogparam" default="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=5   $(arg logdir)" />

  <node pkg="sensor_driver" type="masternode" name="masternode" args="$(arg glogparam)" output="screen">
  	<param name="config_file" value="$(arg configfile)"/>
  </node>
  <group if="1">
  
	  <node pkg="covgrid_slam_ros" type="lidarodometrymain" name="lidarodometrymain" args="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=1   $(arg logdir)" output="screen">
	  <param name="configuration_directory" value="$(find covgrid_slam_ros)/config" />
	  <param name="configuration_basename" value="localmap.lua" />
	  <param name="withoutimu" value="true"/>
	  <param name="withgps" value="true"/>
	  <param name="insvelocity" value="false"/>
	  <param name="withecu" value="false"/>
	  <param name="sendflag" value="false"/>
	  <param name="local_ip" value="192.168.1.112"/>
	  <param name="local_port" value="9911"/>
	  <param name="remote_ip" value="192.168.1.111"/>
	  <param name="remote_port" value="9912"/>
	  </node>
</group>

<group if="1">
  <node pkg="my_map" type="my_map" name="segfusionnode"  output="screen">
  <param name="cloudtopic" value="lidar_cloud_calibrated"/>
  <param name="calibration_mode" value="true"/>
  <param name="pos_x" value="0"/>
  <param name="pos_y" value="1.2088"/>
  <param name="pos_z" value="-0.73"/>
  <param name="roll" value="-0.916638"/>
  <param name="pitch" value="-0.315794"/>
  <param name="yaw" value="0"/>
  </node>
</group>

  <node pkg="sensor_driver" type="getmultilidardata" name="getmultilidardata" args="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=1   $(arg logdir)" output="screen">
  </node>

<group if="$(arg online_mode)">
  <node pkg="sensor_driver" type="getgpsdata" name="getgpsdata" args="$(arg glogparam)" output="screen">
  <remap from="gpsdata" to="GPSmsg" />
  </node>
<node pkg="sensor_driver" type="getinsdata" name="getinsdata" args="$(arg glogparam)" output="screen">
	</node>  
 </group>


  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find covgrid_slam_ros)/rviz_cfg/covgrid_mapping.rviz" />
  </group>

</launch>
